lib/matplotlib/cbook/__init__.py

```diff
@@ -280,6 +280,31 @@ def process(self, s, *args, **kwargs):
                     else:
                         raise
 
+    @contextlib.contextmanager
+    def block(self, *, signal=None):
+        """
+        Block callback signals from being processed.
+
+        A context manager to temporarily block/disable callback signals
+        from being processed by the registered listeners.
+
+        Parameters
+        ----------
+        signal : str, optional
+            The callback signal to block, default is None to block all signals
```

Comment:
```suggestion
        signal : str, default: None
            The callback signal to block, default blocks all signals
```
and this may need an almost doctest style example string? I didn't get how to use this w/o looking at the tests.  Maybe go long form-'signal_name' ?

Reply:
`signal` is used in the process/disconnect functions as well, so I think that we should stick to that here.

I added a quick example of how to use this to the `CallbackRegistry()` docstring. Is that what you were looking for, or something else with the doctest style example?

I added this under the API change note, I am not sure if this is truly a great "next what's new" feature because I didn't expect many external users to use it, but I could move it over if you think that would be a better place for it and add some of the test examples there.

Reply:
I think it'd be a useful short thing to mention in what's new. Plenty of folks build interactive tools & I think the temporary blocking could be useful for 'em.